genisys/server/genisysinventory.py

```diff
@@ -13,11 +14,16 @@ def __init__(self: Self, filepath: str):
         self.filepath = filepath
         self.fd = open(filepath, "r+", encoding="utf-8")
 
-        # Load any host entries from the past
-        try:
-            self.running_inventory = json.load(self.fd)
-        except json.decoder.JSONDecodeError:
+        # Check if json exists
+        if not path.exists(filepath):
+            raise FileNotFoundError
```

Comment:
What's the reason behind raising this? I was under the impression that the `open` call would either create the file or raise its own error if it doesn't exist.

Reply:
The open call uses `"r+"` which doesn't create the file if not found. The arg that creates the file if it does not exist is `"w+"`, but during testing I found out that using `"w+"` automatically overwrites the contents of the file during the open call which deletes the entire inventory file before it can be copied into the running inventory. I can add a clause that makes the directory if it doesn't exist, though.

Reply:
On second thought, I understand what you mean. Removing the check since the `open()` call will error either way. [7de148e](https://github.com/xeluior/genisys/pull/59/commits/7de148e63a946c6af55a07153690f5c5e94a27af)